From: Yong Zhang <yong.zhang0@gmail.com>
Subject: hrtimer: Add missing debug_activate() aid
Date: Thu, 13 Oct 2011 15:52:30 +0800

It will fix below warning, which is also reported by Fernando:

[    7.616090] ------------[ cut here ]------------
[    7.616093] WARNING: at kernel/hrtimer.c:391 hrtimer_fixup_activate+0x27/0x50()
[    7.616094] Hardware name: OptiPlex 755
[    7.616096] Modules linked in:
[    7.616099] Pid: 0, comm: kworker/0:0 Tainted: G        W 3.0.6-rt17-00284-g9d73a61 #15
[    7.616100] Call Trace:
[    7.616103]  [<c014d9a2>] warn_slowpath_common+0x72/0xa0
[    7.616106]  [<c0175417>] ? hrtimer_fixup_activate+0x27/0x50
[    7.616109]  [<c0175417>] ? hrtimer_fixup_activate+0x27/0x50
[    7.616112]  [<c014d9f2>] warn_slowpath_null+0x22/0x30
[    7.616115]  [<c0175417>] hrtimer_fixup_activate+0x27/0x50
[    7.616118]  [<c03b3ab0>] debug_object_activate+0x100/0x130
[    7.616121]  [<c0176b96>] ? hrtimer_start_range_ns+0x26/0x30
[    7.616123]  [<c0175a59>] enqueue_hrtimer+0x19/0x100
[    7.616126]  [<c0176b96>] ? hrtimer_start_range_ns+0x26/0x30
[    7.616129]  [<c0176744>] __hrtimer_start_range_ns+0x144/0x540
[    7.616132]  [<c072705a>] ? _raw_spin_unlock_irqrestore+0x3a/0x80
[    7.616136]  [<c0176b96>] hrtimer_start_range_ns+0x26/0x30
[    7.616139]  [<c01852b5>] tick_nohz_restart_sched_tick+0x185/0x1b0
[    7.616142]  [<c0101878>] cpu_idle+0x98/0xc0
[    7.616146]  [<c071fcd8>] start_secondary+0x1d3/0x1da
[    7.616148] ---[ end trace 0000000000000003 ]---

Reported-by: Fernando Lopez-Lezcano <nando@ccrma.stanford.edu>
Signed-off-by: Yong Zhang <yong.zhang0@gmail.com>
Link: http://lkml.kernel.org/r/20111013075230.GA2740@zhy
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 kernel/hrtimer.c |    1 +
 1 file changed, 1 insertion(+)

Index: linux-stable/kernel/hrtimer.c
===================================================================
--- linux-stable.orig/kernel/hrtimer.c
+++ linux-stable/kernel/hrtimer.c
@@ -1063,6 +1063,7 @@ int __hrtimer_start_range_ns(struct hrti
 			 * remove it again and report a failure. This avoids
 			 * stale base->first entries.
 			 */
+			debug_deactivate(timer);
 			__remove_hrtimer(timer, new_base,
 					timer->state & HRTIMER_STATE_CALLBACK, 0);
 		}
